### PR TITLE
fix playkit embed

### PR DESCRIPTION
**Pull request by OrenMe**
Added fixes to support new syntax of client lib and fix iframe embed

### CODE REVIEW

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 1]
```diff
@@ -156,30 +158,64 @@ private function getOutputHash($o)
...
+		foreach ($config as $key=>$val){
+			$config[$key] = json_decode($val);
+		}
+
+		$config["partnerId"] = $this->partnerId;		
+		$config["uiConfId"] = $this->uiconfId;
+		
+		try{
+			$config = json_encode($config);		
+		} catch (Exception $ex) {
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: json_encode does not throw , on failure it will return false.

Comment (OrenMe) [reply alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: fixed

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 2]
```diff
@@ -156,30 +158,63 @@ private function getOutputHash($o)
...
-		$config = json_encode($this->getRequestParameter("config"));
-		$entry_id = $this->getRequestParameter('entry_id');
-		$autoEmbedCode = "\n var player; var ovpProvider = new Providers.OvpProvider($this->partnerId,\"\",$config);\n" .
-			"\t    ovpProvider.getConfig(\"" . $entry_id . "\",$this->uiconfId).then(config => {\n" .
-			"\t    player = Playkit.playkit(config);\n" .
-			"\t }, \n" .
-			"\t err => {\n" .
-			"\t    console.log(err)\n" .
-			"\t})\n";
+		if (!$targetId){
```
Comment (yossipapi) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: Better do 
$targetId = $targetId ? $targetId : $this->getRequestParameter('targetId');
and than only check if value defined once.

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 3]
```diff
@@ -156,30 +158,63 @@ private function getOutputHash($o)
...
+		foreach ($config as $key=>$val){
+			$config[$key] = json_decode($val);
+		}
+
+		$config["partnerId"] = $this->partnerId;		
+		$config["uiConfId"] = $this->uiconfId;
+		
+		$config = json_encode($config);		
+		if ($config === false){
+			KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, "Invalid config object");
```
Comment (yossipapi) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: The MISSING_PARAMETER error is not correct here more like INVALID_PARAMETER.

File: alpha/apps/kaltura/lib/KExternalErrors.class.php [hunk 4]
```diff
@@ -59,6 +59,7 @@ class KExternalErrors
 	const BUNDLE_CREATION_FAILED = 47;
 	const ENTRY_NOT_SEQUENCE = 48;
 	const INVALID_MIN_BITRATE = 49;
+	const INVALID_PARAMETER = 50;
```
Comment (yossipapi) [inline alpha/apps/kaltura/lib/KExternalErrors.class.php]: You forgot to add this value to the errorDescriptionMap as well.
You can use the [%s] to output a more meaningful error that will reflect what you expected to get.

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 5]
```diff
@@ -156,30 +158,66 @@ private function getOutputHash($o)
...
-		$entry_id = $this->getRequestParameter('entry_id');
-		$autoEmbedCode = "\n var player; var ovpProvider = new Providers.OvpProvider($this->partnerId,\"\",$config);\n" .
-			"\t    ovpProvider.getConfig(\"" . $entry_id . "\",$this->uiconfId).then(config => {\n" .
-			"\t    player = Playkit.playkit(config);\n" .
-			"\t }, \n" .
-			"\t err => {\n" .
-			"\t    console.log(err)\n" .
-			"\t})\n";
+		$targetId = $targetId ? $targetId : $this->getRequestParameter('targetId');
+		$hasTargetId = !is_null($targetId) && $targetId != "";
```
Comment (yossipapi) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: Why add a parameter for the check simply check this "!is_null($targetId) && $targetId != """ inside the if

**Pull request merged**